tests/test_other.py

```diff
@@ -1620,7 +1620,7 @@ def test_js_link(self):
     ''')
     open(os.path.join(self.get_dir(), 'before.js'), 'w').write('''
       var MESSAGE = 'hello from js';
-      if (typeof Module != 'undefined') throw 'This code should run before anything else!';
+      if (Object.keys(Module).length) throw 'This code should run before anything else!';
```

Comment:
This goes into `--pre-js` and while fixing #5900 I've moved `var Module; Module = ...` into `emcc.py` where it now appears before any `--pre-js`.

Because of this change `Module` is now not only defined, but also initialized with empty object, which allows `Module['on_module'] = 'waka';` to work instead of failing with `TypeError: Module is undefined`.

Reply:
I see, thanks. And the `length` check is seeing if the object is empty? Makes sense, but please add a comment explaining it.

Reply:
As I think this is the last change for this PR, please add that comment with `[ci skip]` in the comment, then it skips running all the CI.